openhands/llm/llm.py

```diff
@@ -122,6 +124,12 @@ def __init__(
         if self.is_function_calling_active():
             logger.debug('LLM: model supports function calling')
 
+        # Compatibility flag: use string serializer for DeepSeek models
+        # See this issue: https://github.com/All-Hands-AI/OpenHands/issues/5818
+        self._use_string_serializer = False
+        if 'deepseek' in self.config.model:
+            self._use_string_serializer = True
```

Comment:
I think we don't even need this bit, we can do it only below, when we set this attribute on the Message. IMHO it's better because it's localized in one place. We'll think how to improve it later.

Reply:
I think the trick is that, we only do this string serializer for deepseek because it seems their official API doesn't support list as messages, but we will need to keep the list serializer for other models?

Reply:
Yes, please see below

Reply:
ahhh i didn't see you suggested change